### PR TITLE
Use uv

### PR DESCRIPTION
**Pull request by zsimic**
Added support for `uv`.

### CODE REVIEW

File: src/pickley/__init__.py [hunk 1]
```diff
@@ -1038,3 +1069,26 @@ def represented(self):
...
         return "\n".join(result)
+
+
+def latest_pypi_version(package_name, index):
+    package_name = PypiStd.std_package_name(package_name)
+    if package_name:
+        url = f"{index.rstrip('/')}/{package_name}/"
+        response = http_get(url)
+        if response:
+            versions = sorted(i.version for i in _parsed_simple_html(response) if i.version.is_final)
```
Comment (thatch) [inline src/pickley/__init__.py:1080]: `max` with a try/except might be simpler?

File: src/pickley/__init__.py [hunk 2]
```diff
@@ -7,17 +7,27 @@
...
+    "install_timeout",
+    "package_manager",
+    "preferred_pythons",
+    "python_installations",
+    "pyenv",
+    "version",
+    "version_check_delay",
+}
 PLATFORM = platform.system().lower()
+RX_HREF = re.compile(r'href=".+/([^/#]+\.(tar\.gz|whl))#', re.IGNORECASE)
```
Comment (thatch) [inline src/pickley/__init__.py:30]: This is a non-exhaustive list, in particular .tar.bz2 and .zip are fairly common.  See https://github.com/jwodder/pypi-simple/blob/ba1fec22b4af2294b2cae152342d4bdc3ccd9f94/src/pypi_simple/filenames.py#L10

File: src/pickley/__init__.py (hunk 1 above)
Comment (thatch) [inline src/pickley/__init__.py:1080]: are pre-only projects too much hassle?

**Review by thatch — APPROVED** [review]
sure

File: src/pickley/__init__.py (hunk 2 above)
Comment (zsimic) [reply src/pickley/__init__.py:30]: Would these ever come up as sole artifacts for a CLI? The idea here is to simply parse the `/simple` html for versions (nothing besides the available versions is relevant to pickley)

File: src/pickley/__init__.py (hunk 1 above)
Comment (zsimic) [reply src/pickley/__init__.py:1080]: No pre-release only CLI ever came up so far. Handling pre-releases is possible, but I'd have to define "desired" version better... currently "latest non-pre-release" is the only thing "desired". I do plan to formalize this better in the future, with 3 kinds of "desired track":
- latest (default)
- pre-release (ie: latest not limited to final versions)
- pinned (everything else, tracked correctly, ie an `==` would be an absolute pin, an `~=` would be respected appropriately etc)

Comment (zsimic) [reply src/pickley/__init__.py:1080]: I could change `sorted()` to `max()` indeed here, no need for try/except (as only valid versions are yielded)

Comment (zsimic) [reply src/pickley/__init__.py:1080]: Ah I see what you mean, for an empty `/simple/` response. I'll do it because it adds one more test edge case :)
